compiler/rustc_typeck/src/collect.rs

```diff
@@ -1545,7 +1545,7 @@ fn fn_sig(tcx: TyCtxt<'_>, def_id: DefId) -> ty::PolyFnSig<'_> {
                     let ret_ty = fn_sig.output();
                     if ret_ty != tcx.ty_error() {
```

Comment:
I think this check can now be removed thanks to `is_suggestable` accounting for `Error`.